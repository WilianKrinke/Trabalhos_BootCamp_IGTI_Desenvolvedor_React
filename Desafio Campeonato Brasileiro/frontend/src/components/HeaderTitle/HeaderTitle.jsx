import React from 'react';
import { string } from 'prop-types';

const HeaderTitle = ({title}) => {
    return <h1>{title}</h1>
}

HeaderTitle.propTypes = {
    title: string.isRequired
}

export default HeaderTitle;
