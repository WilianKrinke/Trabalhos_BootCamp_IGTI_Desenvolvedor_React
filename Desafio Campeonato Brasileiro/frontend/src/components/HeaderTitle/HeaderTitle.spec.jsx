import React from "react";
import renderer from 'react-test-renderer'
import HeaderTitle from './HeaderTitle'
import {render, screen} from '@testing-library/react'

describe('HeaderTitle test component', () => {
    it('should render header title', () => {        
        render(<HeaderTitle title={'Desafio React Campeonato Brasileiro'}/>)

        const h1Element = screen.getByText(/desafio react campeonato brasileiro/i)
        expect(h1Element).toBeInTheDocument();
    });

    it('matches snapshot 1', () => {
        const tree = renderer.create(
            <HeaderTitle title={'Desafio React Campeonato Brasileiro'}/>
        ).toJSON()
        expect(tree).toMatchSnapshot();
    });
});
