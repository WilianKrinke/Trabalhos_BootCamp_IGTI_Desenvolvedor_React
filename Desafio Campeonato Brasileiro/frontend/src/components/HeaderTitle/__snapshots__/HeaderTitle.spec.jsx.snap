// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderTitle component matches snapshot 1 1`] = `
<h1>
  Desafio React Campeonato Brasileiro
</h1>
`;

exports[`HeaderTitle test component matches snapshot 1 1`] = `
<h1>
  Desafio React Campeonato Brasileiro
</h1>
`;
